pkg/synthetictests/disruption.go

```diff
@@ -276,3 +277,70 @@ func isOneSecondEvent(eventInterval monitorapi.EventInterval) bool {
 		return true
 	}
 }
+
+func testDnsOverlapDisruption(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
+	const testName = "[sig-network] Disruption should not overlap with DNS problems in cluster running tests"
+	failures := []string{}
+	sortedEvents := mergeIntervals(events)
+
+	for i, event := range sortedEvents {
+		if reason := monitorapi.ReasonFrom(event.Message); strings.HasPrefix(reason, "Disruption") {
+			found := false
+			index := 0
+			for j := i - 1; event.From.Sub(sortedEvents[j].From) < 1*time.Second; j-- {
+				if backenddisruption.DnsLookupRegex.MatchString(sortedEvents[j].Message) {
+					found = true
+					index = j
+					break
+				}
+			}
+
+			if found && event.From.Sub(sortedEvents[index].From) <= 100*time.Second {
+				failures = append(failures, event.From.Format(time.RFC3339)+" "+event.Message)
+			} else if found && event.From.Sub(sortedEvents[index].From) <= 30*time.Second {
+				failures = append(failures, event.From.Format(time.RFC3339)+" "+event.Message)
+			} else if found && event.From.Sub(sortedEvents[index].From) <= 10*time.Second {
+				failures = append(failures, event.From.Format(time.RFC3339)+" "+event.Message)
+			} else if found && event.From.Sub(sortedEvents[index].From) <= 1*time.Second {
+				failures = append(failures, event.From.Format(time.RFC3339)+" "+event.Message)
+			}
```

Comment:
build two lists of intervals, 
if we see Reason/DisruptionSamplerOutageBegan we know it's DNS disruption
if we see reason/DisruptionBegan we know it's "real"

iterate real disruption list (r) and compare each item to every dns disruption (d).
look for:
- r.from < d.from && r.to >= d.to  (r fully overlaps d)
- r.from <= d.from <= r.to (r partially overlaps start of d)
- d.from <= r.from <= d.to (d partially overlaps start of r)
- r.from <= d.to <= r.to (r partially overlaps end of d)
- d.from <= r.to <= d.to (d partially overlaps end of r)
- r.from - d.to <= 10s
- d.from - r.to <= 10s
- r.to - d.from <= 10s
- d.to - r.from <= 10s